examples/cartrade/package.json

```diff
@@ -20,8 +20,8 @@
     "ethereumjs-common": "^1.5.1",
     "ethereumjs-tx": "^2.1.2",
     "express": "~4.16.1",
-    "fabric-ca-client": "~1.4.0",
-    "fabric-network": "~1.4.0",
+    "fabric-ca-client": "^1.4.0",
```

Comment:
@outSH Thank you for the quick fix for the CVE! Could you lock it down to a specific version (in this case it would be `1.4.19`) by any chance? The reason I ask is because having the exact versions specified prevents auto-upgrades and it would get us (closer) to the utopia of [reproducible builds](https://reproducible-builds.org/)

(I'll just comment on this one file of the diff but the same question goes for all the other package.json files of course)

Reply:
@petermetz Builds are already reproducible due to `yarn.lock`, or am I missing something?? In which scenario range definitions are getting in your way?

Reply:
@outSH It would go like this (at least that's how I imagine it, willing to be wrong of course)

1. The example works today with 1.4.19, we all go home happy
2. Next week 1.4.20 gets released that introduces a bug that breaks the example
3. New prospective contributors/users come in, try the example and it's broken because their npm cache did not have 1.4.19 and the range made it so that they got 1.4.20 because they ran the examples from scratch post-release of the 1.4.20 version of the package.

Reply:
@petermetz

> they got 1.4.20 because they ran the examples from scratch post-release of the 1.4.20 version of the package.

But, as long as they use `yarn run configure` as documented, they will still get 1.4.19 even after an update was released. I understand having precise version would be safer for users not familiar with package pinning, but I'm afraid this could be problematic in future update management. I've already encountered problem when trying to upgrade just `fabric-ca-client` with `~1.4.0` specifier, probably due to it's dependencies (didn't investigate it further, though), I'm afraid having plain `1.4.19` could make it worse.

Either way, if you really sure this is the best way to go, I can change that, let me know

Reply:
@outSH 

> But, as long as they use yarn run configure as documented

The cartrade example (and many others) are not part of the monorepo builds (the configure script). I had provided avenue to remedy this in an earlier PR but my work was deleted, so as of right now, it's just dangling outside of the lerna build, same as before.
If you look at the cartrade example readme there is no mention of using yarn configure, it is telling you to switch directories into the example and then run npm install yourself.

> I've already encountered problem when trying to upgrade just fabric-ca-client with ~1.4.0 specifier, probably due to it's dependencies (didn't investigate it further, though)

Hard to tell without further details.

> Either way, if you really sure this is the best way to go, I can change that, let me know

Thank you! I vote for exact versions, regardless of it being more effort to manually bump them later or not.

Reply:
@petermetz Oh, right! I'm currently working on putting them into containers and monorepo build right now, and for some reasons assumed it's already done in my mind :)

> Thank you! I vote for exact versions, regardless of it being more effort to manually bump them later or not.

OK. Shall I switch to exact versions for every dependency in `package.json`? Only fabric one? Shall I do this for all `package.json` affected by this PR?

Reply:
@petermetz nevermind, I've changed only fabric packages to exact versions, since only them are related to this commit, if you wish to continue this process please open new issue describing the scope :)